### PR TITLE
Clotho_v2 Audio Captioning (DCASE 2023 implementation)

### PR DESCRIPTION
**Pull request by Shikhar-S**
## What?
Implementation of the BEATs-BART Audio Captioning system described in [this](https://arxiv.org/pdf/2309.17352) paper.

Our Scores
cider_d: 46.1
spider: 29.8

Paper's comparable scores
cider_d: 45.8
spider: 29.6

(Other numbers for this implementation are in the README and the paper's numbers can be found in second last row of Table 2 (w/o Instructor).)

This PR also implements 1) an option to not load pre-trained weights in ESPnet HuggingFace decoder, and 2) modify the pre-trained model's architecture using external configs.

## Why?

This PR improves on the data preparation (training on all 5 captions, text preprocessing), and implements the same decoder as the paper's implementation (more attention heads, no pre-trained weight loading).

## See also
Builds on top of https://github.com/espnet/espnet/pull/5915

### CODE REVIEW

**Comment by sw005320** [comment]
Thanks, @Shikhar-S 

@ftshijt, can you review this PR again?

File: egs2/clotho_v2/asr1/cmd.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+# If jobs failed, your configuration might be wrong for your environment.
+#
+#
+# The official documentation for run.pl, queue.pl, slurm.pl, and ssh.pl:
+#   "Parallelization in Kaldi": http://kaldi-asr.org/doc/queue.html
+# =========================================================~
+
+
+# Select the backend used by run.sh from "local", "stdout", "sge", "slurm", or "ssh"
+cmd_backend='slurm'
```
Comment (ftshijt) [inline egs2/clotho_v2/asr1/cmd.sh:31]: Please revert back to local (and default setup for slurm configuration, otherwise the ci would complain)

**Review by ftshijt — COMMENTED** [review]
LGTM! thanks for the updates. Please fix the CI issues accordingly~

File: egs2/clotho_v2/asr1/cmd.sh (hunk 1 above)
Comment (Shikhar-S) [reply egs2/clotho_v2/asr1/cmd.sh]: Thanks for catching this. Fixed.

**Comment by Shikhar-S** [comment]
> LGTM! thanks for the updates. Please fix the CI issues accordingly~

Done. Please lmk if any other changes are necessary.

File: egs2/clotho_v2/asr1/run_upload.sh
Comment (sw005320) [inline egs2/clotho_v2/asr1/run_upload.sh:1]: Do you need this file?
It has the absolute path.
We also discussed that we don't need to separate an upload shell.
Can you double-check it?

**Review by sw005320 — COMMENTED** [review]
Can you create a unit test, which covers 1) HF transformers_decoder and 2) beats_encoder? If 1) is tricky, you can try 2) only.

Check https://github.com/espnet/espnet/tree/master/test/espnet2/asr/encoder

File: egs2/clotho_v2/asr1/run_upload.sh
Comment (Shikhar-S) [reply egs2/clotho_v2/asr1/run_upload.sh:1]: Thanks for catching this. Rechecked and fixed.

**Comment by Shikhar-S** [comment]
Handled all comments and added tests.
The ci fails for due to low coverage. However most of the lines it cites are either covered by other tests, or are trivial (like var assignment).

File: egs2/README.md [hunk 2]
```diff
@@ -30,6 +30,7 @@ See: https://espnet.github.io/espnet/espnet2_tutorial.html#recipes-using-espnet2
 | bur_openslr80           | Burmese ASR training dataset                                                                                                     | ASR                     | BUR                   | https://openslr.org/80/                                                                                      |              |
 | catslu               	  | CATSLU-MAPS                                                                                                                      | SLU                     | CMN           	       | https://sites.google.com/view/catslu/home                                                                    |              |
 | catslu_entity        	  | CATSLU                                                                                                                           | SLU/Entity Classifi.    | CMN           	       | https://sites.google.com/view/catslu/home                                                                    |              |
+| clotho_v2                  | Clotho v2.1 dataset for audio captioning | ASR   | ENG                   | https://zenodo.org/records/4783391
```
Comment (sw005320) [inline egs2/README.md:33]: Can you change ASR --> AAC?

**Comment by sw005320** [comment]
Thanks! This is an amazing contribution!

**Pull request merged**